web/src/pages/billing/components/HailBatchGrid.tsx

```diff
@@ -0,0 +1,498 @@
+                return <span>{formattedDuration}</span>
+            },
+        },
+        {
+            category: 'cost',
+            title: 'COST',
+            dataMap: (dataItem: any, _value: string) => (
+                <Popup
+                    content={dataItem.cost}
+                    trigger={<span>${dataItem.cost.toFixed(4)}</span>}
+                    position="top center"
+                />
+            ),
+        },
+    ]
+
+    const DETAIL_FIELDS: Field[] = [
+        {
+            category: 'topic',
+            title: 'TOPIC',
+        },
+        {
+            category: 'namespace',
+            title: 'NAMESPACE',
+        },
+        {
+            category: 'batch_name',
+            title: 'NAME/SCRIPT',
+        },
+        {
+            category: 'job_name',
+            title: 'NAME',
+        },
+    ]
+
+    const expandedRow = (log: any, idx: any) =>
+        MAIN_FIELDS.map(({ category, dataMap, className }) => (
+            <SUITable.Cell key={`${category}-${idx}`} className={className}>
+                {dataMap ? dataMap(log, log[category]) : sanitiseValue(log[category])}
+            </SUITable.Cell>
+        ))
+
+    return (
+        <Table celled compact sortable>
+            <SUITable.Header>
+                <SUITable.Row>
+                    <SUITable.HeaderCell style={{ borderBottom: 'none' }} />
+                    {MAIN_FIELDS.map(({ category, title }, i) => (
+                        <SUITable.HeaderCell
+                            key={`${category}-${i}`}
+                            style={{
+                                borderBottom: 'none',
+                                position: 'sticky',
+                                resize: 'horizontal',
+                                textAlign: 'center',
+                            }}
+                        >
+                            {title}
+                        </SUITable.HeaderCell>
+                    ))}
+                </SUITable.Row>
+                <SUITable.Row>
+                    <SUITable.Cell
+                        style={{
+                            borderTop: 'none',
+                            backgroundColor: 'var(--color-table-header)',
+                        }}
+                    />
+                    {MAIN_FIELDS.map(({ category }, i) => (
+                        <SUITable.Cell
+                            className="sizeRow"
+                            key={`${category}-resize-${i}`}
+                            style={{
+                                borderTop: 'none',
+                                backgroundColor: 'var(--color-table-header)',
+                            }}
+                        ></SUITable.Cell>
+                    ))}
+                </SUITable.Row>
+            </SUITable.Header>
+            <SUITable.Body>
+                {combinedData
+                    .sort((a, b) => {
+                        if (a.batch_id < b.batch_id) {
```

Comment:
Can you provide a human readable string, eg:

```suggestion
                        // Sorts an array of objects first by 'batch_id' and then by 'job_id' in ascending order.
                        if (a.batch_id < b.batch_id) {
```